library/src/main/java/org/apache/apex/malhar/lib/wal/FileSystemWAL.java

```diff
@@ -680,9 +681,7 @@ protected void flush() throws IOException
           //until the stream is closed on the local FS, readers don't see any data.
           close();
         } else {
-          Syncable syncableOutputStream = (Syncable)outputStream;
-          syncableOutputStream.hflush();
-          syncableOutputStream.hsync();
+          ((HdfsDataOutputStream)outputStream).hsync(EnumSet.of(SyncFlag.UPDATE_LENGTH));
```

Comment:
If we don't specify flag and don't use HDFS dependency, During recovery, code should not rely on the file length returned by the APIs as namenode has wrong file length, this behavior is confusing.  Ideally, HSYNC should have updated the NameNode, so I consider passing the flag as a cleaner approach, as it updates the Namenode and the data is persisted.

@PramodSSImmaneni please let me know your take as well.